app/controllers/users_controller.rb

```diff
@@ -19,10 +21,30 @@ def show
     @user = User.find(params[:id])
   end
 
+  def new_with_invitation_token
+    invitation = Invitation.find_by(token: params[:token])
+    if invitation
+      @user = User.new(name: invitation.recipient_name, email: invitation.recipient_email)
+      @invitation_token = invitation.token
+    else
+      @user = User.new
+    end
+    render :new
+  end
+
   private
 
   def user_params
     params.require(:user).permit(:email, :password, :name)
   end
 
+  def handle_invitation
+    if params[:invitation_token].present?
+      invitation = Invitation.find_by(token: params[:invitation_token])
+      @user.follow(invitation.inviter)
+      invitation.inviter.follow(@user)
+      invitation.update_column(:token, nil)
```

Comment:
I would rather see a model method for this part `.update_column(:token, nil)` something like `.remove_token` and it could be refactored in future with `token` related methods.